<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from es165_macro.xacro              | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="es165d">
  <!-- link list -->
  <link name="base_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://motoman_es_support/meshes/es/visual/base_link.stl"/>
      </geometry>
      <material name="blue">
        <color rgba="0 0 1 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://motoman_es_support/meshes/es/collision/base_link.stl"/>
      </geometry>
      <material name="yellow">
        <color rgba="0 1 1 1"/>
      </material>
    </collision>
  </link>
  <link name="link_s">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://motoman_es_support/meshes/es/visual/link_s.stl"/>
      </geometry>
      <material name="blue"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://motoman_es_support/meshes/es/collision/link_s.stl"/>
      </geometry>
      <material name="yellow"/>
    </collision>
  </link>
  <link name="link_l">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://motoman_es_support/meshes/es/visual/link_l.stl"/>
      </geometry>
      <material name="blue"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://motoman_es_support/meshes/es/collision/link_l.stl"/>
      </geometry>
      <material name="yellow"/>
    </collision>
  </link>
  <link name="link_u">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://motoman_es_support/meshes/es/visual/link_u.stl"/>
      </geometry>
      <material name="blue"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://motoman_es_support/meshes/es/collision/link_u.stl"/>
      </geometry>
      <material name="yellow"/>
    </collision>
  </link>
  <link name="link_r">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://motoman_es_support/meshes/es165/visual/link_r.stl"/>
      </geometry>
      <material name="blue"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://motoman_es_support/meshes/es165/collision/link_r.stl"/>
      </geometry>
      <material name="yellow"/>
    </collision>
  </link>
  <link name="link_b">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://motoman_es_support/meshes/es/visual/link_b.stl"/>
      </geometry>
      <material name="blue"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://motoman_es_support/meshes/es/collision/link_b.stl"/>
      </geometry>
      <material name="yellow"/>
    </collision>
  </link>
  <link name="link_t">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://motoman_es_support/meshes/es/visual/link_t.stl"/>
      </geometry>
      <material name="black">
        <color rgba="0 0 0 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://motoman_es_support/meshes/es/collision/link_t.stl"/>
      </geometry>
      <material name="black"/>
    </collision>
  </link>
  <link name="tool0"/>
  <!--

      Balancer and rod are disabled as without a mimic joint they will not update
      their pose properly, leading to MoveIt detecting collisions which cannot
      happen with the real robot (as the balancer is essentially a passive element
      and cannot come into collision with anything due to joint limits).

    <link name="${prefix}balancer_piston">
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <mesh filename="package://motoman_es_support/meshes/es/visual/balancer_piston.stl"/>
        </geometry>
        <material name="blue"/>
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <mesh filename="package://motoman_es_support/meshes/es/collision/balancer_piston.stl"/>
        </geometry>
        <material name="yellow"/>
      </collision>
    </link>
    <link name="${prefix}balancer_rod">
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <mesh filename="package://motoman_es_support/meshes/es/visual/balancer_rod.stl"/>
        </geometry>
        <material name="blue">
          <color rgba="1 1 1 1"/>
        </material>
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <mesh filename="package://motoman_es_support/meshes/es/collision/balancer_rod.stl"/>
        </geometry>
        <material name="yellow"/>
      </collision>
    </link>
    -->
  <!-- end of link list -->
  <joint name="joint_1_s" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.450"/>
    <parent link="base_link"/>
    <child link="link_s"/>
    <axis xyz="0 0 1"/>
    <limit effort="0" lower="-3.141592653589793" upper="3.141592653589793" velocity="1.5707963267948966"/>
  </joint>
  <joint name="joint_2_l" type="revolute">
    <origin rpy="0 0 0" xyz="0.740 0 0"/>
    <parent link="link_s"/>
    <child link="link_l"/>
    <axis xyz="0 1 0"/>
    <limit effort="0" lower="-2.2689280275926285" upper="1.3962634015954636" velocity="1.4835298641951802"/>
  </joint>
  <joint name="joint_3_u" type="revolute">
    <origin rpy="0 0 0" xyz="1.150 0 0"/>
    <parent link="link_l"/>
    <child link="link_u"/>
    <axis xyz="0 -1 0"/>
    <limit effort="0" lower="-1.9547687622336491" upper="3.6302848441482056" velocity="1.4835298641951802"/>
  </joint>
  <joint name="joint_4_r" type="revolute">
    <origin rpy="0 0 0" xyz="0.250 0 0"/>
    <parent link="link_u"/>
    <child link="link_r"/>
    <axis xyz="0 0 1"/>
    <limit effort="0" lower="-6.283185307179586" upper="6.283185307179586" velocity="3.0543261909900767"/>
  </joint>
  <joint name="joint_5_b" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 -1.225"/>
    <parent link="link_r"/>
    <child link="link_b"/>
    <axis xyz="0 -1 0"/>
    <limit effort="0" lower="-2.2689280275926285" upper="2.2689280275926285" velocity="2.6179938779914944"/>
  </joint>
  <joint name="joint_6_t" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 -0.225"/>
    <parent link="link_b"/>
    <child link="link_t"/>
    <axis xyz="0 0 1"/>
    <limit effort="0" lower="-6.283185307179586" upper="6.283185307179586" velocity="4.1887902047863905"/>
  </joint>
  <joint name="joint_t_tool0" type="fixed">
    <origin rpy="0 3.141592653589793 0" xyz="0 0 0"/>
    <parent link="link_t"/>
    <child link="tool0"/>
  </joint>
  <!--

      See earlier comment about balancer and piston.

    <joint name="${prefix}joint_balancer_piston" type="fixed">
      <origin xyz="-0.260 0 0.320" rpy="0 0 0"/>
      <parent link="${prefix}link_s"/>
      <child link="${prefix}balancer_piston"/>
    </joint>
    <joint name="${prefix}joint_balancer_rod" type="fixed">
      <origin xyz="${0.740 + 0.220 * cos(radians(55))} 0 ${0.220 * sin(radians(55))}" rpy="0 0 0"/>
      <parent link="${prefix}link_s"/>
      <child link="${prefix}balancer_rod"/>
    </joint>
    -->
  <!-- end of joint list -->
  <!-- ROS base_link to Robot Manufacturer World Coordinates transform -->
  <link name="world"/> 
  <joint name="world_fixed_joint" type="fixed">
    <parent link="world"/>
    <child link="base_link"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
  </joint>
</robot>

